tests/test_datasets.py

```diff
@@ -1,74 +1,108 @@
+    assert isinstance(dataset.target_names, list)
 
 
-def test_load_moscow_stjoes_as_frame():
+@pytest.mark.parametrize("configuration", CONFIGURATIONS, ids=CONFIGURATION_IDS)
+def test_load_dataset_as_frame(configuration: DatasetConfiguration):
     """Test that the dataset is loaded as a dataframe."""
-    moscow = load_moscow_stjoes(as_frame=True)
+    dataset = configuration.load_function(as_frame=True)
 
-    assert isinstance(moscow.frame, pd.DataFrame)
-    assert isinstance(moscow.data, pd.DataFrame)
-    assert isinstance(moscow.target, pd.DataFrame)
+    assert isinstance(dataset.frame, pd.DataFrame)
+    assert isinstance(dataset.data, pd.DataFrame)
+    assert isinstance(dataset.target, pd.DataFrame)
 
-    assert moscow.data.shape == (165, 28)
-    assert moscow.target.shape == (165, 35)
-    assert moscow.frame.shape == (165, 28 + 35)
-    assert_array_equal(moscow.frame.index.values, moscow.index)
+    assert dataset.data.shape == (configuration.n_samples, configuration.n_features)
+    assert dataset.target.shape == (configuration.n_samples, configuration.n_targets)
+    assert dataset.frame.shape == (
+        configuration.n_samples,
+        configuration.n_features + configuration.n_targets,
+    )
+    assert_array_equal(dataset.frame.index.values, dataset.index)
 
 
-def test_load_moscow_stjoes_as_xy():
+@pytest.mark.parametrize("configuration", CONFIGURATIONS, ids=CONFIGURATION_IDS)
+def test_load_dataset_as_xy(configuration: DatasetConfiguration):
     """Test that the dataset is loaded as X y arrays."""
-    X, y = load_moscow_stjoes(return_X_y=True)
-    assert X.shape == (165, 28)
-    assert y.shape == (165, 35)
+    X, y = configuration.load_function(return_X_y=True)
+    assert X.shape == (configuration.n_samples, configuration.n_features)
+    assert y.shape == (configuration.n_samples, configuration.n_targets)
 
 
-def test_load_moscow_stjoes_as_xy_as_frame():
+@pytest.mark.parametrize("configuration", CONFIGURATIONS, ids=CONFIGURATION_IDS)
+def test_load_dataset_as_xy_as_frame(configuration: DatasetConfiguration):
     """Test that the dataset is loaded as X y dataframes."""
-    X, y = load_moscow_stjoes(return_X_y=True, as_frame=True)
-    data = load_moscow_stjoes()
+    X, y = configuration.load_function(return_X_y=True, as_frame=True)
+    data = configuration.load_function()
 
     assert isinstance(X, pd.DataFrame)
     assert isinstance(y, pd.DataFrame)
-    assert X.shape == (165, 28)
-    assert y.shape == (165, 35)
+    assert X.shape == (configuration.n_samples, configuration.n_features)
+    assert y.shape == (configuration.n_samples, configuration.n_targets)
     assert_array_equal(X.index.values, data.index)
     assert_array_equal(y.index.values, data.index)
 
 
-def test_asframe_raises_without_pandas():
+@pytest.mark.parametrize("configuration", CONFIGURATIONS, ids=CONFIGURATION_IDS)
+def test_asframe_raises_without_pandas(configuration: DatasetConfiguration):
```

Comment:
Perhaps not necessary to test both configurations on this one - I don't know why one would raise and the other wouldn't.  But I was doing blanket replacement and chose to do this one as well.

Reply:
Yeah, not necessary but it doesn't do any harm. I'm good with that choice.